File: back/src/main/java/com/tdevilleduc/urthehero/back/model/NextPage.java
package com.tdevilleduc.urthehero.back.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.validation.constraints.NotNull;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
@EqualsAndHashCode
public class NextPage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @NotNull
    @JsonIgnore
    private Integer id;
    @NotNull
    private String text;
    @NotNull
    private Integer pageId;
    @NotNull
    private Integer destinationPageId;
    @NotNull
    private Position position;

}

File: back/src/main/java/com/tdevilleduc/urthehero/back/model/NextPage.kt
package com.tdevilleduc.urthehero.back.model

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id

@Entity
data class NextPage(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @JsonIgnore
        var id: Int? = null,
        var text: String = "",
        var pageId: Int = 0,
        var destinationPageId: Int = 0,
        var position: Position? = null

)